extension/json/src/functions/table_functions/json_scan.cpp

```diff
@@ -97,25 +142,63 @@ struct JSONScanLocalState : public TableFuncLocalState {
     uint8_t* bufferPtr = nullptr;
     idx_t bufferOffset = 0;
     uint64_t bufferSize = 0;
+    uint64_t bufferStartByteOffsetInFile = 0;
     uint64_t numValuesToOutput = 0;
     storage::MemoryManager& mm;
+    idx_t lineCountInBuffer;
+    std::unique_ptr<processor::LocalFileErrorHandler> errorHandler;
 
-    JSONScanLocalState(storage::MemoryManager& mm, BufferedJsonReader* reader)
-        : docs{}, currentReader{reader},
+    JSONScanLocalState(storage::MemoryManager& mm, JSONScanSharedState& sharedState,
+        main::ClientContext* context)
+        : docs{}, currentReader{sharedState.jsonReader.get()},
           reconstructBuffer{
               mm.allocateBuffer(false /* initializeToZero */, JsonConstant::SCAN_BUFFER_CAPACITY)},
-          mm{mm} {}
+          mm{mm}, lineCountInBuffer(0),
+          errorHandler(std::make_unique<processor::LocalFileErrorHandler>(
+              &sharedState.sharedErrorHandler, false, context)) {}
+
+    ~JSONScanLocalState() override {
```

Comment:
I would suggest moving the definition after the class declaration

Reply:
updated